libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -64,6 +62,9 @@ class MapboxNavigation(
                 ::paramsProvider
             )
         )
+        directionsSession.registerRouteObserver(createInternalRouteObserver())
+        directionsSession.registerRouteObserver(navigationSession)
```

Comment:
Where are we unregistering these observers registered inside `init`

Reply:
I'm going to cut a ticket to audit the code and check that all of the observers registered get unregistered. This is not only here 👍

Reply:
We need to audit. This case should be safe though as the passed arguments and the sessions have the same lifecycle (all are created by `MapboxNavigation`).

Reply:
Follow up ticket https://github.com/mapbox/mapbox-navigation-android/issues/2418